Slim/Schema.pm

```diff
@@ -3293,6 +3300,22 @@ sub canFulltextSearch {
 	return $canFulltextSearch;
 }
 
+sub _workRequired {
+	my $genres = shift;
+	return 1 if !$prefs->get('worksScan');
+	return 0 if $prefs->get('worksScan') == 2;
+	my $genreMatch = 0;
+	my %scanGenres = map {$_ => 1} split(/,\s*/, uc($prefs->get('myClassicalGenres')));
```

Comment:
I'm probably a bit crazy about re-using code. But we're doing the exact same thing in `Releases.pm`: split the uppercased pref, create a map, then check whether something (upper case) is in that map.

Shouldn't this be implemented once as eg. `isClassicalGenre($genre)`? 

You should even keep the map around so you don't have to rebuild it all the time: this will be called tens or hundreds of thousands of times during a scan. Create a `setChange` handler to reset the list when the pref changes.